app/javascript/controllers/unassign_evaluator_submission_modal_controller.js

```diff
@@ -0,0 +1,72 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["modal", "confirmButton"]
+  static values = {
+    phaseId: String,
+    assignmentId: String
+  }
+
+  connect() {
+    this.modalTarget.addEventListener('click', this.handleOutsideClick.bind(this))
+  }
+
+  disconnect() {
+    this.modalTarget.removeEventListener('click', this.handleOutsideClick.bind(this))
+  }
+
+  open(event) {
+    event.preventDefault();
+    this.setValues(event.currentTarget.dataset);
+    this.modalTarget.showModal();
+  }
+
+
+  close() {
+    this.modalTarget.close()
+  }
+
+  handleOutsideClick(event) {
+    if (event.target === this.modalTarget) {
+      this.close()
+    }
+  }
+
+  confirm() {
+    this.unassignEvaluatorSubmission()
+  }
+
+  setValues(dataset) {
+    this.assignmentIdValue = dataset.assignmentId;
+    this.phaseIdValue = dataset.phaseId;
+  }
+
+  unassignEvaluatorSubmission() {
+    const csrfToken = document.querySelector('meta[name="csrf-token"]').content;
+    
+    fetch(`/phases/${this.phaseIdValue}/evaluator_submission_assignments/${this.assignmentIdValue}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': csrfToken,
+        'Accept': 'application/json'
+      },
+      body: JSON.stringify({
+        status: 'unassigned'
```

Comment:
I tried modifying this value to send some unknown, invalid status to the backend to trigger an error but it caused a 500 exception instead in the model update. You can remedy this by validating the status param in the rails controller before trying to update the model with an invalid value. if the client sends an invalid status, reject the request with `:unprocessable_entity` status.

Reply:
.. alternatively you could rescue errors from the `@assignment.update` call in the controller to return false